app/src/main/java/com/demoapp/database/DataRepository.kt

```diff
@@ -1,16 +1,17 @@
 package com.demoapp.database
 
-import android.util.Log
+
 import com.demoapp.model.CategoryEntity
 import com.demoapp.model.DataEntity
 import com.demoapp.model.DataResponse
 import com.demoapp.retrofit.RetrofitInterface
-import retrofit2.Call
-import retrofit2.Callback
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
 
-class DataRepository(val dataDuo: DataDuo, val mService: RetrofitInterface) {
+class DataRepository @Inject constructor(val dataDuo: DataDuo, val mService: RetrofitInterface) {
 
-    fun getLocalData(result : (List<DataEntity>) -> Unit) {
+    suspend fun getLocalData(): List<DataEntity> {
         var resultList = arrayListOf<DataEntity>()
```

Comment:
Instead of additional list variable like this. use array map function to transform one data type to another

Reply:
Removed extra variable and returned list with the use of arrayListOf().apply